pinot-common/src/main/java/org/apache/pinot/common/utils/config/TierConfigUtils.java

```diff
@@ -76,20 +76,33 @@ public static boolean shouldRelocateToTiers(TableConfig tableConfig) {
   }
 
   /**
-   * Comparator for sorting the {@link Tier}.
-   * As of now, we have only 1 type of {@link TierSegmentSelector} and 1 type of {@link TierStorage}.
-   * Tier with an older age bucket in {@link TimeBasedTierSegmentSelector} should appear before a younger age bucket,
-   * in sort order
-   * TODO: As we add more types, this logic needs to be upgraded
+   * Comparator for sorting the {@link Tier}. In the sort order,
+   * 1) {@link FixedTierSegmentSelector} are always before others
+   * 2) For {@link TimeBasedTierSegmentSelector}, tiers with an older age bucket appear before a younger age bucket,
    */
   public static Comparator<Tier> getTierComparator() {
     return (o1, o2) -> {
       TierSegmentSelector s1 = o1.getSegmentSelector();
       TierSegmentSelector s2 = o2.getSegmentSelector();
-      Preconditions.checkState(TierFactory.TIME_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s1.getType()),
+
+      Preconditions.checkState(TierFactory.TIME_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s1.getType())
+              || TierFactory.FIXED_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s1.getType()),
           "Unsupported segmentSelectorType class %s", s1.getClass());
-      Preconditions.checkState(TierFactory.TIME_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s2.getType()),
+      Preconditions.checkState(TierFactory.TIME_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s2.getType())
+          || TierFactory.FIXED_SEGMENT_SELECTOR_TYPE.equalsIgnoreCase(s2.getType()),
           "Unsupported segmentSelectorType class %s", s2.getClass());
```

Comment:
(minor) These checks are redundant. We don't really need to validate the `type` of a selector